Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -10,243 +10,424 @@
+  if (atom_indexes.size() != 4) {
+    return;
+  }
 
-    if(pdb_info.m_residue_name && pdb_info.m_residue_name->isDefined(atom_num))
-    {
-        rd_info->setResidueName(pdb_info.m_residue_name->at(atom_num));
-    }
+  auto *bond = mol.getBondBetweenAtoms(atom_indexes[1], atom_indexes[2]);
+  CHECK_INVARIANT(bond, "bad stereo bond");
+  CHECK_INVARIANT(bond->getBondType() == Bond::DOUBLE, "bad stereo bond");
 
-    if(pdb_info.m_chain_id && pdb_info.m_chain_id->isDefined(atom_num))
-    {
-        rd_info->setChainId(pdb_info.m_chain_id->at(atom_num));
+  bond->setStereoAtoms(atom_indexes[0], atom_indexes[3]);
+  bond->setStereo(type);
+}
+
+//! Copy over the structure properties, including stereochemistry.
+void set_mol_properties(RWMol &mol, const mae::Block &ct_block) {
+  for (const auto &prop : ct_block.getProperties<std::string>()) {
+    if (prop.first == mae::CT_TITLE) {
+      mol.setProp(common_properties::_Name, prop.second);
+    } else if (prop.first.find(mae::CT_CHIRALITY_PROP_PREFIX) == 0 ||
+               prop.first.find(mae::CT_PSEUDOCHIRALITY_PROP_PREFIX) == 0) {
+      parseChiralityLabel(mol, prop.second);
+    } else if (prop.first.find(mae::CT_EZ_PROP_PREFIX) == 0) {
+      parseStereoBondLabel(mol, prop.second);
+    } else {
+      mol.setProp(prop.first, prop.second);
     }
+  }
+  for (const auto &prop : ct_block.getProperties<double>()) {
+    mol.setProp(prop.first, prop.second);
+  }
+  for (const auto &prop : ct_block.getProperties<int>()) {
+    mol.setProp(prop.first, prop.second);
+  }
+  for (const auto &prop : ct_block.getProperties<mae::BoolProperty>()) {
+    mol.setProp(prop.first, static_cast<bool>(prop.second));
+  }
+}
 
-    if(pdb_info.m_insertion_code &&
-            pdb_info.m_insertion_code->isDefined(atom_num))
-    {
-        rd_info->setInsertionCode(pdb_info.m_insertion_code->at(atom_num));
+//! Set atom properties. Some of these have already been parsed to construct the
+//! Atom object, and should be skipped. Also, atom properties may be undefined
+//! for an atom, and should also be skipped for that atom.
+void set_atom_properties(Atom &atom, const mae::IndexedBlock &atom_block,
+                         size_t i) {
+  for (const auto &prop : atom_block.getProperties<std::string>()) {
+    if (prop.first == PDB_ATOM_NAME || prop.first == PDB_RESIDUE_NAME ||
+        prop.first == PDB_CHAIN_NAME || prop.first == PDB_INSERTION_CODE) {
+      // PDB information is parsed separately.
+      continue;
+    } else if (!prop.second->isDefined(i)) {
+      continue;
     }
 
-    if(pdb_info.m_resnum && pdb_info.m_resnum->isDefined(atom_num))
-    {
-        rd_info->setResidueNumber(pdb_info.m_resnum->at(atom_num));
+    atom.setProp(prop.first, prop.second->at(i));
+  }
+
+  for (const auto &prop : atom_block.getProperties<double>()) {
+    if (prop.first == mae::ATOM_X_COORD || prop.first == mae::ATOM_Y_COORD ||
+        // Coordinates are used in defining a conformation, and should not be
```

Comment:
weird place for the comment, please move it either into the block or out of it.

Reply:
Yes, that one is quite weird. Not sure how it got there, though.